syntax = "proto3";

service PeerService {
    rpc GetPortForUser(PortForUserRequest) returns (PortResponse) {}
    rpc SetUserPort(UserPortRequest) returns (Response) {}
    rpc GetFile(FileRequest) returns (FileResponse) {}
    rpc CreateFile(FileCreateRequest) returns (FileCreateResponse) {}
}

message PortForUserRequest {
    string user = 1;
}

message PortResponse {
    string port = 1;
}

message UserPortRequest {
    string user = 1;
}

message FileRequest {
    string filename = 1;
}

message FileResponse {
    string content = 1;
}

message FileCreateRequest {
    repeated string filename = 1;
}

message Response{
    bool success = 1;
}

message FileCreateResponse {
    bool success = 1;
}
